### ISSUE TITLE
Oxlint nested `.oxlintrc.json` is not applied when it runs `oxlint` from workspace root.

### ISSUE THREAD

**Issue by emosheeep** [oxc-project/oxc#22867]
## Environment

- `oxlint`: `1.67.0`
- VS Code extension: `oxc.oxc-vscode`
- OS: Linux

## Repository structure

```text
repo/
├── package.json
├── node_modules/
│   └── @byted-image/oxlint/
│       └── configs/
│           └── ts.json
└── packages/
    └── lv-infra-plugins/
        ├── .oxlintrc.json
        └── rule-test.ts
```

`packages/lv-infra-plugins/.oxlintrc.json`:

```json
{
  "extends": [
    "../../node_modules/@byted-image/oxlint/configs/ts.json"
  ],
  "ignorePatterns": [
    "types/**"
  ]
}
```

The extended config enables `oxc/no-const-enum`.

`packages/lv-infra-plugins/rule-test.ts`:

```ts
const enum Enum {
  A = 'A',
  B = 'B',
  C = 'C',
}
```

### 1. Run Oxlint from the package directory

```sh
cd packages/lv-infra-plugins
oxlint rule-test.ts --format json
```

`oxc/no-const-enum` is reported.

### 2. Run Oxlint from the repo root with the same file path

```sh
oxlint packages/lv-infra-plugins/rule-test.ts --format json
```

`oxc/no-const-enum` is not reported.

### 3. Run Oxlint from the repo root with explicit config

```sh
oxlint \
  -c packages/lv-infra-plugins/.oxlintrc.json \
  packages/lv-infra-plugins/rule-test.ts \
  --format json
```

`oxc/no-const-enum` is reported.

### 4. Open the file in VS Code

Open `packages/lv-infra-plugins/rule-test.ts` with the `oxc.oxc-vscode` extension enabled.

`oxc/no-const-enum` is not reported in the editor, matching the root CLI behavior.

## Expected behavior

The same file should produce the same diagnostics in these cases:

1. package directory CLI
2. repo root CLI
3. VS Code extension

In this example, `oxc/no-const-enum` should be reported consistently.

**Comment by camc314** [comment]
Hi!

Thanks for reporting this issue.

I tried to reproduce the issue here: https://github.com/camc314/oxlint-repros/tree/main/issue-22867 exactly how you described.

However, in my testing, all three commands, as well as via the editor reported the `no-const-enum` error.

It is possible that there is something else in your repo that is affecting this setup - please PR to that folder if this is the case.

Thanks!

**Comment by emosheeep** [comment]
<img width="2148" height="646" alt="Image" src="https://github.com/user-attachments/assets/1be51704-9032-4fe2-ba81-8ba2d6c62ddc" />

I reduced the repro further and found the missing trigger.

The previous minimal repro did not reproduce because the shared config only had:

```json
{
  "rules": {
    "oxc/no-const-enum": "error"
  }
}
```

If I add `plugins: ["typescript"]` to the same extended config, the mismatch appears:

```json
{
  "plugins": ["typescript"],
  "rules": {
    "oxc/no-const-enum": "error"
  }
}
```

With this change:

```sh
# Run from repo root
npx oxlint@1.67.0 packages/lv-infra-plugins/rule-test.ts --format json
# only reports eslint(no-unused-vars), exits 0

# Run from package directory
cd packages/lv-infra-plugins
npx oxlint@1.67.0 rule-test.ts --format json
# reports eslint(no-unused-vars) and oxc(no-const-enum), exits 1

# Run from repo root with explicit config
npx oxlint@1.67.0 \
  -c packages/lv-infra-plugins/.oxlintrc.json \
  packages/lv-infra-plugins/rule-test.ts \
  --format json
# reports eslint(no-unused-vars) and oxc(no-const-enum), exits 1
```

So the minimal difference is that the extended config contains `plugins: ["typescript"]`.
Without that field, the root invocation also reports `oxc(no-const-enum)`.
With that field, the root invocation misses it, while package-cwd and explicit-config invocations still report it.

**Comment by emosheeep** [comment]
@camc314 https://github.com/camc314/oxlint-repros/pull/5 Sorry for waiting so long, I just got free and created a PR, please take a check.

**Comment by emosheeep** [comment]
Hi there, did you reproduce the problem? Is there anything I can help?

**Comment by camc314** [comment]
Looking

**Comment by camc314** [comment]
Apologies for the delay!

Thanks for the PR with the updated case - it helped to pin down what was going on.

For more context, previously, the nested config was being discovered, but the default plugin set was not being applied in the same way vs the root config or with `-c`.

The trigger is the extended config having:

```json
{
  "plugins": ["typescript"],
  "rules": {
    "oxc/no-const-enum": "error"
  }
}
```

Since the child `.oxlintrc.json` omits `plugins`, it should still get Oxlint’s default plugins. 

#22903 is merging now which should fix this assumption.

To unblock you in the interim, you can include `oxc` explicitly in the shared config:

```json
{
  "plugins": ["typescript", "oxc"],
  "rules": {
    "oxc/no-const-enum": "error"
  }
}
```

That should make the root CLI, package-cwd CLI, and editor behavior consistent until the fix is released.

This fix should release on monday - thanks for reporting!
